test/utils/asserts.hpp

```diff
@@ -1,7 +1,23 @@
 #pragma once
 
+#include <gmock/gmock.h>
 #include <gtest/gtest.h>
 
+MATCHER(Trapped, "")
+{
+    return arg.trapped == true;
+}
+
+MATCHER(Result, "")
```

Comment:
```suggestion
MATCHER(Result, "empty result")
```

In case of failure this will display 

> Expected: empty result

instead of 

> Expected: result

Not critical, but might be helpful.